python/lsst/ts/lasertracker/laser_tracker_csc.py

```diff
@@ -653,19 +631,60 @@ async def measure_alignment(self, target: str) -> None:
             target=target_frame_name, reference_pointgroup=reference_frame_name
         )
 
-        # Remove Zeropoint from offset
+        # Remove Zeropoint from offset and add LUT
         if target == "M2":
             target_offset["dX"] -= self.config.zero_points["m2"]["x"]
             target_offset["dY"] -= self.config.zero_points["m2"]["y"]
             target_offset["dZ"] -= self.config.zero_points["m2"]["z"]
             target_offset["dRX"] -= self.config.zero_points["m2"]["u"]
             target_offset["dRY"] -= self.config.zero_points["m2"]["v"]
+            # Add LUT
+            compensation_lut = compensation.Compensation(
+                elevation_rotation_coeffs=self.config.targets_lut_coeffs["M2"]["elevation_rotation_coeffs"],
+                azimuth_coeffs=self.config.targets_lut_coeffs["M2"]["azimuth_coeffs"],
+                temperature_coeffs=self.config.targets_lut_coeffs["M2"]["temperature_coeffs"],
+                min_temperature=self.config.min_temperature,
+                max_temperature=self.config.max_temperature,
+            )
+            compensation_inputs = base.CompensationInputs(
+                elevation=self.elevation,
+                azimuth=self.azimuth,
+                rotation=self.camrot,
+                temperature=0.0,
+            )
+            compensation_offsets = compensation_lut.get_offset(compensation_inputs)
+            target_offset["dX"] -= compensation_offsets.x * 1e-3
+            target_offset["dY"] -= compensation_offsets.y * 1e-3
+            target_offset["dZ"] -= compensation_offsets.z * 1e-3
+            target_offset["dRX"] -= compensation_offsets.u
+            target_offset["dRY"] -= compensation_offsets.v
         elif target == "CAM":
             target_offset["dX"] -= self.config.zero_points["camera"]["x"]
             target_offset["dY"] -= self.config.zero_points["camera"]["y"]
             target_offset["dZ"] -= self.config.zero_points["camera"]["z"]
             target_offset["dRX"] -= self.config.zero_points["camera"]["u"]
             target_offset["dRY"] -= self.config.zero_points["camera"]["v"]
+            # Add LUT
+            compensation_lut = compensation.Compensation(
+                elevation_rotation_coeffs=self.config.targets_lut_coeffs["CAM"]["elevation_rotation_coeffs"],
+                azimuth_coeffs=self.config.targets_lut_coeffs["CAM"]["azimuth_coeffs"],
```

Comment:
Tiago's point is that you don't need this part within the if statement. Just put it before the if statement or after, and change "CAM" for the variable `target`